Sources/Async/Socket/SocketSink.swift

```diff
@@ -33,28 +26,28 @@ public final class SocketSink<Socket>: InputStream
     /// True if we are waiting on upstream
     private var isAwaitingUpstream: Bool
 
+    /// Creates a new `SocketSink`
     internal init(socket: Socket, on worker: Worker) {
         self.socket = socket
         self.eventLoop = worker.eventLoop
         self.inputBuffer = nil
-        self.written = 0
         self.isAwaitingUpstream = false
-        let writeSource = self.eventLoop.onWritable(descriptor: socket.descriptor, writeSourceSignal)
+        let writeSource = self.eventLoop.onWritable(
+            descriptor: socket.descriptor,
+            // config: .init(trigger: .level),
+            writeSourceSignal
+        )
         writeSource.resume()
         self.writeSource = writeSource
     }
 
     /// See InputStream.input
     public func input(_ event: InputEvent<UnsafeBufferPointer<UInt8>>) {
+        // update variables
         switch event {
         case .next(let input):
             isAwaitingUpstream = false
-
-            /// crash if the upstream is illegally overproducing data
-            guard inputBuffer == nil else {
-                fatalError("\(#function) was called while inputBuffer is not nil")
-            }
-
+            assert(inputBuffer == nil, "SocketSink upstream is illegally overproducing input buffers.")
```

Comment:
Why doesn't this crash anymore? It could have quite some concequences if this is ignored